cloudinit/net/network_manager.py

```diff
@@ -71,6 +71,57 @@ def _set_default(self, section, option, value):
         if not self.config.has_option(section, option):
             self.config[section][option] = value
 
+    def _config_option_is_set(self, section, option):
+        """
+        Checks if a config option is set. Returns True if it is,
+        else returns False.
+        """
+        return self.config.has_section(section) and self.config.has_option(
+            section, option
+        )
+
+    def _get_config_option(self, section, option):
+        """
+        Returns the value of a config option if its set,
+        else returns None.
+        """
+        if self._config_option_is_set(section, option):
+            return self.config[section][option]
+        else:
+            return None
+
+    def _change_set_config_option(self, section, option, value):
+        """
+        Overrides the value of a config option if its already set.
+        Else, if the config option is not set, it does nothing.
+        """
+        if self._config_option_is_set(section, option):
+            self.config[section][option] = value
+
+    def _set_mayfail_true_if_both_false_dhcp(self):
+        """
+        If for both ipv4 and ipv6, 'may-fail' is set to be False,
+        set it to True for both of them.
+        """
+        for family in ["ipv4", "ipv6"]:
+            if self._get_config_option(family, "may-fail") != "false":
```

Comment:
I agree that this check is currently not required. But I still want to have this explicitly spelled out.